src/decode-vxlan.c

```diff
@@ -0,0 +1,276 @@
+/* Copyright (C) 2019 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+/**
+ * \file
+ *
+ * \author Henrik Kramshoej <hlk@kramse.org>
+ *
+ * VXLAN decoder.
+ */
+
+#include "suricata-common.h"
+#include "decode.h"
+#include "decode-vxlan.h"
+#include "decode-events.h"
+
+#include "detect-engine-port.h"
+
+#include "flow.h"
+
+#include "util-unittest.h"
+#include "util-debug.h"
+
+#include "pkt-var.h"
+#include "util-profiling.h"
+#include "host.h"
+
+#define VXLAN_HEADER_LEN        8
+#define VXLAN_DEFAULT_PORT      4789
+#define VXLAN_DEFAULT_PORT_S    "4789"
+
+static bool g_vxlan_enabled = true;
```

Comment:
I wonder if we should disable by default, as this is a backport to stable. Thoughts @jasonish ?

Reply:
I'm in favour of having it disabled by default in 4.1.[](url)

Reply:
@shivan1b can you make sure it is disabled by default? That means this global should be set to false, and the suricata.yaml.in setting should also be updated. It will there also require an update to the suricata-verify tests (add --set decoder.vxlan.enabled=true to the commandline I think)